### ISSUE TITLE
[Dungeon Finder] Queueing multiple times with a group can lead to dungeons not resetting

### ISSUE THREAD

**Issue by syssneck** [azerothcore/azerothcore-wotlk#8222]
### Current Behaviour

Queueing multiple times with a group (without disbanding in between the runs) can lead to dungeons not resetting. We got teleported to the same dungeon that we just did (BRD) and we immediately saw, that the mobs at the entrance were missing.

Somehow we didn't get a new ID through the dungeon finder and got teleported to the "old" dungeon. Upon disbanding the group (because we did this dungeon already) we got the deserter debuff.

### Expected Blizzlike Behaviour

Dungeons that were queued and entered through the dungeon finder should always be "new" (reset).

### Steps to reproduce the problem

1. run a lot of rdfs with a group
2. have bad luck?
3. and see that the dungeon didn't reset
(rare bug, hard to reproduce)

### AC rev. hash/commit

6c953e9333d8ff23

### Operating system

Win 10 x64

**Comment by UltraNix** [comment]
Yep - need more info.

**Comment by syssneck** [comment]
Yes, it was finished in the sense of rdf (also got satchel). We got teleported to BRD - Prison entrance, but all the mobs and also Interrogator were not there (bodies despawned), so there was no way to finish the rdf. This was the second time it happened (the other one was also with BRD).

**This happened before this issue occurred:**
We only queued as two and got BRD. We killed Trash & Interrogator and two of the group decided to stick around for more runs, so we queued again as 4. I accidentally clicked on decline, so we had to wait for a short amount of time and then queued again. We got instant rdf pop and got teleported to BRD - Prison again, but the mobs that we killed before were missing. Other Trash groups & bosses are all present. Only what we killed before was missing.

This is all the info I can give for now. This bug never happens, when we disband the group and reinvite. I will update if I find out more.

**Comment by UltraNix** [comment]
@syssneck But all players were in the same group and instance and can see each other?

**Comment by syssneck** [comment]
Yes, we were all there

**Comment by UltraNix** [comment]
@syssneck Poke me with all details  if happen again.

**Comment by Nefertumm** [comment]
Confirmed by multiple people. One guy just told us it happened when a last player stays at the dungeon and he joins the group to another dungeon via LFD without leaving the dungeon that the group just did.

**Comment by UltraNix** [comment]
@Nefertumm That another dungeon is a just new group?

**Comment by Nefertumm** [comment]
No, with the same group

**Comment by Nefertumm** [comment]
It seems it happened with another guy who just join the LFD alone.

**Comment by UltraNix** [comment]
@Nefertumm The guy that just joined the LFD was in that dungeon already or somewhere in the open world?

**Comment by Nefertumm** [comment]
@UltraNix he doesn't know for sure, but he is 70% he was doing quests while it happened.

**Comment by UltraNix** [comment]
And does he remember who was the leader after teleport to that new dungeon? Maybe it was him?

**Comment by Nefertumm** [comment]
He doesn't remember that, sadly :/ I'll get more reports for this, as I'm in touch with a bunch of players of LATAM, I'll let you know on discord or here.

**Comment by UltraNix** [comment]
Poke me on discord

**Comment by UltraNix** [comment]
@Luckdemon2 
`It put me back in the same dungeon that had already been cleared out` And what happened to others? You used random DF this time again or maybe specific and got luck to get the same dungeon?
